lib/utils/analytics/generatePayload.js

```diff
@@ -85,5 +93,11 @@ module.exports = async (serverless) => {
     npmDependencies,
     triggeredDeprecations: Array.from(triggeredDeprecations),
     versions,
+    createdAt: Date.now(),
```

Comment:
This imposes the problem of having wrong timestamps being send to our database. As we do not have guarantee that user has proper time set on their machine. It's easy that way to send data from the future.

So ideally timestamps should be guarded at our servers. Technically we could just set them on our server side, but there's one issue. Analytics as send from the framework while in most cases is about deploys made at very given moment, can also send data about deployment made up to one week before (if intially analytics request won't send successfully it is cached and it's reattempted to send within a one week time frame, if that time passes, event is discarded from cache and not send).

So I guess, we can keep sending timestamp here, but on our side (in metrics service) I think we need to validate that it's within `oneWeekBefore > timestamp < now` time frame, and if it's not then round it to nearest boundary.

Reply:
That is a great point - my reasoning here was to capture the timestamp at which event was generated - at first I wanted to generate it only server-side in backend service, but then I realised that these events can be cached and sent with a delay - so I think the best way to move forward will be to keep it being generated here but add "normalization" logic on the backend, correct?

Reply:
Yes, exactly. As you had same in mind, my comment was not needed 👍

Reply:
It was much needed, as I didn't think of potential drift of time or misconfigured time on user side so thanks for bringing that up - I'll address it on backend side :+1: